syne_tune/blackbox_repository/conversion_scripts/scripts/tabrepo_import.py

```diff
@@ -0,0 +1,270 @@
+        print(f"Processing dataset: {dataset_name}")
+        hyperparameters_configurations = {}
+        for framework in group.index.to_frame(index=False)["framework"].values:
+            hyperparameters_configurations[framework] = repo.config_hyperparameters(
+                config=framework
+            )
+        bb_dict[dataset_name] = convert_dataset(
+            config_space, group, hyperparameters_configurations
+        )
+
+    with catchtime("saving to disk"):
+        serialize(
+            bb_dict=bb_dict,
+            path=repository_path / bb_name,
+            metadata={metric_elapsed_time: METRIC_ELAPSED_TIME},
+        )
+
+
+def convert_dataset(config_space, evaluations, configurations):
+    # names of hyperparameters
+    hp_cols = list(config_space.keys())
+    # number of hyperparameters
+    n_hps = len(hp_cols)
+    # number of evaluations
+    n_evals = len(evaluations.xs(0, level="fold"))
+    # initialize hyperparameter array with shape(n_evals, n_hps)
+    hps = np.zeros((n_evals, n_hps), dtype=object)
+
+    n_seeds = 3
+
+    for i, config in enumerate(configurations):
+        arr = []
+        for key in config_space.keys():
+            # this handles cases, where not all hps are set
+            if key in configurations[config].keys():
+                # this avoids type errors in fastparquet conversion, applies only to RandomForrest and ExtraTrees
+                if key == "max_features":
+                    arr.append(str(configurations[config][key]))
+                else:
+                    arr.append(configurations[config][key])
+            # set hp to None if not existent
+            else:
+                arr.append(None)
+        hps[i] = arr
+
+    hyperparameters = pd.DataFrame(data=hps, columns=hp_cols)
+    # can I just rename time_train to metric elapsed time or does the data need to be changed somehow?
+    objective_names = [
+        "metric_error",
+        "metric_error_val",
+        "metric_elapsed_time",
+        "metric_time_infer_s",
+        "metric_rank",
+    ]
+    n_objectives = len(objective_names)
+
+    objective_evaluations = []
+
+    for i, fold in enumerate([0, 1, 2]):
+        fold_data = evaluations.xs(fold, level="fold")
+        for j, (_, row) in enumerate(fold_data.iterrows()):
+            metrics = row[
+                [
+                    "metric_error",
+                    "metric_error_val",
+                    "time_train_s",
+                    "time_infer_s",
+                    "rank",
+                ]
+            ].values
+
+            if len(objective_evaluations) <= j:
+                objective_evaluations.append([[], [], []])
+
+            objective_evaluations[j][i] = metrics
+    objective_evaluations = np.array(objective_evaluations)
+    objective_evaluations = objective_evaluations.reshape(
+        n_evals, n_seeds, 1, n_objectives
+    )
+    # fidelity space initialized as constant value, since it is required as an argument
```

Comment:
```suggestion
    # Tabrepo does not provide performance across different fidelities. We initialize the fidelity space as a constant value, since it is required as an argument
```